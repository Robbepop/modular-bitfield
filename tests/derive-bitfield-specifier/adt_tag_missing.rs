use modular_bitfield::prelude::*;

#[derive(BitfieldSpecifier)]
pub enum Two {
    Zero,
    One,
    Two,
    Three,
}

#[derive(BitfieldSpecifier)]
pub enum Tag {
    First,
    Second,
    ThisIsOk,
    AlsoOk,
}

#[derive(BitfieldSpecifier)]
#[tag(Tag)]
pub enum TagMissing {
    First(Two, Two),
    Second(Two),
    Third,
    Fourth,
}

fn main() {}
