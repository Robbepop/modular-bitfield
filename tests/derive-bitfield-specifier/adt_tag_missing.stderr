error[E0599]: no variant or associated item named `Third` found for enum `Tag` in the current scope
  --> tests/derive-bitfield-specifier/adt_tag_missing.rs:24:5
   |
12 | pub enum Tag {
   | ------------ variant or associated item `Third` not found here
...
24 |     Third,
   |     ^^^^^ variant or associated item not found in `Tag`

error[E0599]: no variant or associated item named `Fourth` found for enum `Tag` in the current scope
  --> tests/derive-bitfield-specifier/adt_tag_missing.rs:25:5
   |
12 | pub enum Tag {
   | ------------ variant or associated item `Fourth` not found here
...
25 |     Fourth,
   |     ^^^^^^ variant or associated item not found in `Tag`
